src/components/ProductList/ProductList.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+
+  <ul class="items-container"> 
+    <ProductListItem v-for="item in items" 
+    :id="item.id" 
+    :key="item.id"
+    :title="item.title" 
+    :url="item.url"
+    :price="item.price"
+     @delete-item="openDeleteWindow"
+     @editing-item="openEditWindow"
+    />
+      <Modal 
+     v-if="showModal"
+    :initialTitle="item.title"
+    :initialPrice="item.price"
+    :initialUrl="item.url"
+    :itemId="item.id"
+    :title="action"
+    @close-modal="handleClose"
+    >
+    <div slot="header">
+      <p>{{action}}</p>
+    </div>
+    <div  slot="modal-body">
+      <template v-if="action === 'Deleting'">
+        <p>Deleted item will be lost!</p>
+      </template>
+      <template v-else>
+       <add-edit-form 
+       :action="action" 
+       :item="item"
+       @close-modal="handleClose"
+       @save-edit="saveEditedItem"
+       ></add-edit-form>
+      </template>
+    </div>
+    <div slot="modal-footer"> 
+       <template v-if="action === 'Deleting'">
+         <VButton @click="handleClose" text="Cancel"/>
+        <VButton class="button-delete"  @click="handleDelete" text="Delete"/>
+      </template>
+    </div>
+    </Modal>
+      
+  </ul>
+
+ 
+</template>
+
+<script>
+import Modal from "../Modal/Modal.vue"
+import ProductListItem from "../ProductListItem/ProductListItem.vue"
+import VButton from "../VButton/VButton.vue"
+import AddEditForm from "../AddEditForm/AddEditForm.vue"
+import {deleteData} from "../../api/apiCalls"
+
+export default {
+  components: { ProductListItem, Modal, VButton, "add-edit-form": AddEditForm },
+  data() {
+    return {
+      items: [],
+      showModal: false,
+      action: null,
+      itemToDeleteId: null,
+      itemToEditId: null,
+      item: {},
+    }
+  },
+  methods: {
+ 
+    async handleDelete () {
+      const oldDtata = [...this.items]
+      deleteData("shop", `${this.itemToDeleteId}`)
```

Comment:
You are initiating an async call, but proceeding to execute further logic without receiving a success response from the BE/promise.  
Consider a case where BE would be down or error out when delete endpoint is called, it would fail, but UI would still update to reflect the fact that item was deleted.  
Always await the response from endpoint until proceeding with some sort of further logic and be sure to handle endpoint errors.  
Also, template string seems to be used without purpose.  
```suggestion
      deleteData("shop", this.itemToDeleteId)
```